myriad-scheduler/src/main/java/org/apache/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -83,6 +85,31 @@ public void onEvent(ResourceOffersEvent event, long sequence, boolean endOfBatch
     }
     LOGGER.debug("Received offers {}", offers.size());
     LOGGER.debug("Pending tasks: {}", this.schedulerState.getPendingTaskIds());
+
+    // Let Myriad use reserved resources firstly.
+    Collections.sort(offers, new Comparator<Offer>() {
+      boolean isReserved(Offer o) {
+        for (Protos.Resource resource : o.getResourcesList()) {
+          if (resource.hasRole() && !Objects.equals(resource.getRole(), DEFAULT_ROLE)) {
```

Comment:
You shouldn't be comparing against the default role if we allow users to specify a custom role. If a user specifies a non-default role, this code will ignore all resources reserved for it.
But Mesos will never offer resources for a role your framework is not registered for, so checking the value of the role is unnecessary anyway. Just bubbling up any offers with a role set is sufficient for this.